### DISCUSSION TITLE
Adding perturbation to a single variable after a certain number of time steps

### DISCUSSION THREAD

**Discussion by alimostafavi24** [idaholab/moose#24963] [Q&A General]
Hello everyone,

Is there any way in MOOSE to add perturbation to a variable by multiplying it to a perturbation function?

For example, I want to run the simulation for 100 time steps. At time step 101, I want to multiply a variable by a perturbation function and then continue the normal simulation.

Thanks,
Ali

**Comment by GiudGiud** [comment]
Hello

Is it a nonlinear variable (one you solve for) or an auxiliary variable (one you just compute with an auxkernel)

Guillaume

**Reply by alimostafavi24** [reply]
It is a nonlinear variable.

To be more specific, I want to multiply x velocity and z velocity by the perturbation function, just in a single time step, and in the middle of simulation.

Thanks,

**Reply by GiudGiud** [reply]
So the cannon way to do that is unfortunately not great, you d have to make a new UO that changes the dof values. It wont be hard to write if you are keen on C++

However, if you want to stay on the input file scale, use a multiapp with the same mesh. Send your variable there to an auxvariable, multiply it there using a ParsedAux, then send it back using another transfer

**Reply by alimostafavi24** [reply]
Thank you so much.

I also have another approach in my mind. Is it possible to run for 100 time steps and then stop the simulation, then restart the simulation from time step 101, get initial condition for variables from the previous time step, and add the perturbation function in the initial condition block?

**Reply by GiudGiud** [reply]
sure. There's a SumIC I think for example
